src/Calculator/Views/GraphingCalculator/EquationInputArea.xaml.cpp

```diff
@@ -67,8 +69,16 @@ void EquationInputArea::AddNewEquation()
         Equations->GetAt(Equations->Size - 1)->IsLastItemInList = false;
     }
 
-    m_lastLineColorIndex = (m_lastLineColorIndex + 1) % AvailableColors->Size;
-    auto eq = ref new EquationViewModel(ref new Equation(), ++m_lastFunctionLabelIndex, AvailableColors->GetAt(m_lastLineColorIndex)->Color);
+    // Cap equations at number of available colors
+    if (Equations->Size >= m_AvailableColors->Size)
+    {
+        return;
+    }
+
+    m_lastLineColorIndex = (m_lastLineColorIndex + 1) % colorAssignmentMapping.size();
+
+    auto eq =
+        ref new EquationViewModel(ref new Equation(), ++m_lastFunctionLabelIndex, AvailableColors->GetAt(colorAssignmentMapping[m_lastLineColorIndex])->Color);
```

Comment:
I'm afraid it will launch a IndexOutOfRangeException in High Contrast because of the limited number of colors?

Reply:
I confirm, it launches an exception after you add the third equation

Reply:
Good catch!